helpers.go

```diff
@@ -0,0 +1,33 @@
+package leaderelection
+
+import (
+	"fmt"
+	"net"
+)
+
+// GetSelfHostPort gets the global unicast address of where the application is
+// running and returns a slice of IP:Port which can be passed to Config.HostPort
+func GetSelfHostPort(port string) ([]string, error) {
+	addresses, addrErr := net.InterfaceAddrs()
+	if addrErr != nil {
+		return nil, fmt.Errorf("Unable to get self IP address: %w", addrErr)
+	}
+
+	var selfIP []string
```

Comment:
Hmm, we'll almost never return 0 entries. (and returning a `nil` slice can be a bit rude in public APIs)
Let's pre-allocate this with capacity 1 and length 0.